spec/rubocop/cop/lint/number_conversion_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::NumberConversion do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense when using `#to_i`' do
+    expect_offense(<<-RUBY.strip_indent)
+      '10'.to_i
+      ^^^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using "10".to_i, use stricter Integer("10").
+    RUBY
+  end
+
+  it 'registers an offense when using `#to_f`' do
+    expect_offense(<<-RUBY.strip_indent)
+      '10.2'.to_f
+      ^^^^^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using "10".to_i, use stricter Integer("10").
```

Comment:
Hm. This example uses `'10.2'.to_f`. It is not `'10'.to_i`. It would be easier for users to understand a message to use `Float('10.2')` instead of `'10.2'to_f`.

Reply:
Even I think so, in order to accomplish to this, we have to make warning message dynamic correct ?

Reply:
Yeah, the message should be made dynamic - you're just have to interpolate the offending code in the message.